src/main/scala/io/ilyamor/ks/snapshot/Snapshoter.scala

```diff
@@ -300,26 +151,24 @@ case class Snapshoter[S <: Segment, Store <: AbstractRocksDBSegmentedBytesStore[
       case Some(localCheckPointFile) =>
         localCheckPointFile.read().asScala.values.headOption match {
           case Some(offset) =>
-            withLatencyMetrics(
-              downloadStateSensor,
-              () =>
-                storageClientForStore.getStateStores(
-                  context.taskId.toString,
-                  storeName,
-                  context.applicationId(),
-                  offset.toString
-                )
-            ).tapError { e =>
-                downloadStateErrorSensor.record()
+            storageClientForStore
+              .getStateStores(
+                context.taskId.toString,
+                storeName,
+                context.applicationId(),
+                offset.toString
+              )
+              .time(metrics.downloadStateSensor)
```

Comment:
סתם שאלה על סקלה
`.time`
מופעל על קריאה של מתודה או בפועל מופעל כבר על תוצאה, כלומר במקרה שלנו על
 `either`
?

Reply:
השאלה הייתה בעצם על מה מופעל implicit class
על מתודה (פונקציה) או על either בפועל. זה לא ברור מהסבר שלהם. אבל אם אתה בדקת/השתמשת בצורה כזו - אז אני מאמין לך

Reply:
"they won’t be evaluated at all if they are unused. This is similar to replacing the by-name parameters with the passed expressions. They are in contrast to by-value parameters. To make a parameter called by-name"
this is the diffrence between passing by value any by name, if it was by value it was an either, but it passed by name, the same way that zio is lazy, its not eveluated
`  final def apply[A](a: => A): Task[A] = ZIO.apply(a)
`

`    Task(throw new Exception("Unspecified config location"))` will not be evaluated, only when executed later on

Reply:
I know it and initially i did it exactly same. When you have `b.method().time` it seems reasonably, but i came to part where
`methodA().methodB().methodC().time`, so what part `time` is related to. And if it's related to `methodA()` what should I do if i want to measure `time` only for part of `methodB().methodC()`?

Reply:
i am guess you are wrapping everything in some monad, so if you if you do a.b.c.time
b is actually somekind of a flatmap operator(filter or etc) so this will massure
a.b.c
if you do
a.time.b
it will messure a